nixos/tests/ec2.nix

```diff
@@ -0,0 +1,77 @@
+{ system ? builtins.currentSystem }:
+
+with import ../lib/testing.nix { inherit system; };
+with import ../lib/qemu-flags.nix;
+with pkgs.lib;
+
+let
+  image =
+    (import ../lib/eval-config.nix {
+      inherit system;
+      modules = [ ../maintainers/scripts/ec2/amazon-hvm-config.nix ];
+    }).config.system.build.amazonImage;
+
+  makeEc2Test = { name, userData, script, hostname ? "ec2-instance", sshPublicKey ? null }:
+    let
+      metaData = pkgs.stdenv.mkDerivation {
+        name = "metadata";
+        buildCommand = ''
+          mkdir -p $out/2011-01-01
+          ln -s ${pkgs.writeText "userData" userData} $out/2011-01-01/user-data
+
+          mkdir -p $out/1.0/meta-data
+          echo "${hostname}" > $out/1.0/meta-data/hostname
+        '' + optionalString (sshPublicKey != null) ''
+          mkdir -p $out/1.0/meta-data/public-keys/0
+          ln -s ${pkgs.writeText "sshPublicKey" sshPublicKey} $out/1.0/meta-data/public-keys/0/openssh-key
+        '';
+      };
+    in makeTest {
+      name = "ec2-" + name;
+      nodes = {};
+      testScript =
+        ''
+          use File::Temp qw/ tempfile /;
+          my ($fh, $filename) = tempfile();
+
+          `qemu-img create -f qcow2 -o backing_file=${image}/nixos.img $filename`;
+
+          system("cd ${metaData} && ${pkgs.python}/bin/python -m SimpleHTTPServer 27182 &");
```

Comment:
This is our "instance metadata service". It doesn't die when the test ends, unfortunately. I didn't want to spin up a whole separate VM just to configure a proper httpd with NixOS config. Is there a better way to get a lightweight HTTP server that dies properly?